packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -140,17 +139,19 @@ class _InternalSpriteAnimationWidgetState
   }
 
   void _setupController() {
-    _controller?.dispose();
+    widget.animation.onComplete = _pauseAnimation;
+    _controller ??= AnimationController(vsync: this)
+      ..removeListener(_onAnimationValueChanged)
```

Comment:
```suggestion
    _controller ??= AnimationController(vsync: this);
    _controller
      ..removeListener(_onAnimationValueChanged)
```
Solving @wolfenrain's concern